cclient/response/get_balances.c

```diff
@@ -10,25 +10,24 @@
 get_balances_res_t* get_balances_res_new() {
   get_balances_res_t* res =
       (get_balances_res_t*)malloc(sizeof(get_balances_res_t));
-
-  res->balances = int_array_array_new();
-  res->milestone = flex_hash_array_new();
+  if (res) {
```

Comment:
is it possible that this function will return `retcode_t` and return OOM if malloc fails?
or do you check it on the caller side?

Reply:
IMO, it's no difference in returning retcode_t or pointer, both need to check OOM but in the different aspect in this case. If you prefer retcode_t we can do it in all similar functions in another PR to make sure the consistency.

Reply:
ok, cool